src/Noise/InterpolNoise.h

```diff
@@ -348,13 +348,14 @@ class cInterpolNoise
 		ASSERT(a_StartY < a_EndY);
 
 		// Calculate the integral and fractional parts of each coord:
-		int FloorX[MAX_SIZE];
-		int FloorY[MAX_SIZE];
+		int FloorX[MAX_SIZE] = {};
+		int FloorY[MAX_SIZE] = {};
 		NOISE_DATATYPE FracX[MAX_SIZE];
 		NOISE_DATATYPE FracY[MAX_SIZE];
 		int SameX[MAX_SIZE];
 		int SameY[MAX_SIZE];
 		int NumSameX, NumSameY;
+
 		CalcFloorFrac(a_SizeX, a_StartX, a_EndX, FloorX, FracX, SameX, NumSameX);
```

Comment:
These are not good changes. They hurt performance by needlessly filling the arrays with zeroes and then immediately overwriting the values with the proper calculated numbers.
Also the `CalcFloorFrac` call is tightly coupled to the declarations, the additional empty line breaks this logical tie.

As @mjagdis already pointed out, this needs a deeper fix, which he even provided, so please remove this from the PR.